### PR TITLE
chore: fix apidocs 3/3

### PR DESCRIPTION
**Pull request by tomfrenken**
In this PR the remaining missing links are fixed.
Most of it was cross-module references to types/interfaces/functions.

You can ignore all files in the `docs` directory, as they were auto-generated again.

Closes SAP/cloud-sdk-backlog#708.

### CODE REVIEW

File: packages/connectivity/src/scp-cf/destination/destination-cache.ts [hunk 1]
```diff
@@ -211,7 +211,7 @@ async function cacheRetrievedDestination<T extends DestinationCacheInterface>(
  * Sets the custom destination cache instance.
  * Call this method with an instance of {@link DestinationCacheInterface} to override the default cache instance set by the SDK.
  *
- * NOTE: This function should be called at the beginning before any calls to either {@link getDestination} or {@link executeHttpRequest}.
+ * NOTE: This function should be called at the beginning before any calls to either {@link getDestination} or {@link @sap-cloud-sdk/http-client!executeHttpRequest}.
```
Comment (deekshas8) [inline packages/connectivity/src/scp-cf/destination/destination-cache.ts:214]: [q] I notice we use `@link` differently here, could you explain why `{@link executeHttpRequest}` doesn't work?

File: packages/connectivity/src/index.ts [hunk 2]
```diff
@@ -48,7 +48,10 @@ export {
   DestinationJson,
   DestinationsByType,
   ServiceBinding,
-  DestinationForServiceBindingOptions
+  DestinationForServiceBindingOptions,
```
[the remark this answers is not in the document]
Comment (tomfrenken) [reply packages/connectivity/src/index.ts:51]: Sounds like we could remove one of the two.

File: packages/odata-v4/src/de-serializers/entity-deserializer.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { DeSerializers } from './de-serializers';
 
 /**
  * Entity deserializer instance for v4 entities.
- * See {@link EntityDeserializer} for the provided methods.
+ * See {@link @sap-cloud-sdk/odata-common!EntityDeserializer} for the provided methods.
```
[the remark this answers is not in the document]
Comment (tomfrenken) [reply packages/odata-v4/src/de-serializers/entity-deserializer.ts:11]: Good catch, I'll check all occurances.

File: packages/http-client/src/http-client.ts [hunk 4]
```diff
@@ -388,7 +388,7 @@ export function executeHttpRequest<
  * @param destination - A destination or a destination name and a JWT.
  * @param requestConfig - Any object representing an HTTP request.
  * @param options - An {@link HttpRequestOptions} of the HTTP request for configuring e.g., CSRF token delegation. By default, the SDK will fetch the CSRF token.
- * @returns A promise resolving to an {@link HttpResponse}.
+ * @returns A promise resolving to an {@link @sap-cloud-sdk/http-client!HttpResponse}.
```
[the remark this answers is not in the document]
Comment (tomfrenken) [reply packages/http-client/src/http-client.ts:391]: Indeed, I think there were many references from other modules to HttpResponse, so this one was overcorrected 😅

[the remark this answers is not in the document]
Comment (marikaner) [reply packages/http-client/src/http-client.ts:391]: [req] Then please remove it :)

File: packages/odata-v4/src/request-builder/get-by-key-request-builder.ts [hunk 5]
```diff
@@ -17,8 +17,8 @@ import { responseDataAccessor } from './response-data-accessor';
 
 /**
  * Create an OData request to get a single entity based on its key properties.
- * The properties available in the response can be restricted by creating a {@link GetByKeyRequestBuilderV4.select selection}, where no selection is equal to selecting all fields of the entity.
- * Navigational properties need to expanded explicitly by {@link GetAllRequestBuilderV4.expand}
+ * The properties available in the response can be restricted by creating a {@link GetByKeyRequestBuilder.select | selection}, where no selection is equal to selecting all fields of the entity.
+ * Navigational properties need to expanded explicitly by {@link GetAllRequestBuilder.expand}
```
Comment (marikaner) [inline packages/odata-v4/src/request-builder/get-by-key-request-builder.ts:21]: [req] typo
```suggestion
 * Navigational properties need to be expanded explicitly by {@link GetAllRequestBuilder.expand}.
```

File: packages/odata-common/src/entity-builder.ts [hunk 6]
```diff
@@ -87,7 +87,7 @@ export class EntityBuilder<
 
   /**
    * Builds an entity from JSON representation.
-   * If you have obtained the JSON as a request payload use the {@link deserializeEntity} methods.
+   * If you have obtained the JSON as a request payload use the {@link entityDeserializer} methods.
```
Comment (marikaner) [inline packages/odata-common/src/entity-builder.ts:90]: [pp] I think without object this doesn't make sense.

```suggestion
   * If you have obtained the JSON object as a request payload use the {@link entityDeserializer} methods.
```

**Review by marikaner — COMMENTED** [review]
LGTM, I'll leave it to @deekshas8 to approve.

File: packages/connectivity/src/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomfrenken) [reply packages/connectivity/src/index.ts:51]: Looking further into it, it looks like they are still somewhat different, as one of them is more opinionated, I would keep it in for now, but we should change one of their names in the future.

**Review by deekshas8 — APPROVED** [review]
LGTM.

**Pull request merged**